core/src/main/java/de/danoeh/antennapod/core/storage/DBTasks.java

```diff
@@ -289,9 +291,14 @@ public static void notifyMissingFeedMediaFile(final Context context, final FeedM
      */
     public static Future<?> autodownloadUndownloadedItems(final Context context) {
         Log.d(TAG, "autodownloadUndownloadedItems");
-        return autodownloadExec.submit(ClientConfig.automaticDownloadAlgorithm
-                .autoDownloadUndownloadedItems(context));
+        return autodownloadExec.submit(downloadAlgorithm.autoDownloadUndownloadedItems(context));
+    }
 
+    /**
+     * For testing purpose only.
+     */
+    public static void setDownloadAlgorithm(ApDownloadAlgorithm newDownloadAlgorithm) {
```

Comment:
`@VisibleForTesting`? :)

Reply:
Thanks for the hint, I didn't know that annotation. Code is updated.